File: mario.c
#include <stdio.h>
#include <cs50.h>

void get_height(void);
void print_row(int len);

int height;

int main(void)
{
    //Get height
    get_height();

    //Print pyramid
    print_row(height);
}


//Get height function
void get_height(void)
{
    //Get height
    height = get_int("Height: ");

    //Check if height is between 1 and 8
    if (!(height >= 1 && height <= 8))
    {
        get_height();
    }
}

void print_row(int len)
{
    //Return if len of row is 0
    if (len == 0)
    {
        return;
    }

    //Print shorter row first
    print_row(len - 1);

    //Print spaces
    for (int i = 0; i < height - len; i++)
    {
        printf(" ");
    }

    //Print bricks
    for (int i = 0; i < len; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: mario.py
from cs50 import get_int


# Main function
def main():
    # Get height
    height = get_height()

    # Print pyramid
    print_pyramid(height)


# Get height function
def get_height():
    # Get height
    height = get_int("Height: ")

    # Check if height is between 1 and 8
    if height < 1 or height > 8:
        return get_height()

    # Return height
    return height


# Print pyramid function
def print_pyramid(h, l=None):
    # If length is None set it equal to height
    if l == None:
        l = h

    # Stop if length is 0
    if l == 0:
        return

    # Print shorter row first
    print_pyramid(h, l - 1)

    # Print spaces
    print(" " * int(h - l), end="")

    # Print bricks(left side)
    print("#" * int(l), end="")
    
    # Print two spaces
    print("  ", end="")
    
    # Print bricks(right side)
    print("#" * int(l), end="")

    # Start a new line
    print()


# Call main function
if __name__ == "__main__":
    main()
